tests/drivers/fprotect/negative/src/main.c

```diff
@@ -88,19 +88,19 @@ static void test_flash_write_protected_fails(void)
 	flash_write_protected_fails(TEST_FPROTECT_WRITE_ADDR, true);
 }
 
-static void test_flash_write_protected_unmodified(void)
+ZTEST(test_fprotect_negative, test_flash_write_protected_unmodified)
 {
 	flash_write_protected_unmodified(TEST_FPROTECT_WRITE_ADDR);
 }
 
-static void test_bootloader_protection(void)
+ZTEST(test_fprotect_negative, test_bootloader_protection)
 {
 #ifdef CONFIG_SECURE_BOOT
```

Comment:
You can consider using in this test `Z_TEST_SKIP_IFNDEF`:
https://github.com/zephyrproject-rtos/zephyr/blob/bb6fa5707f77d72307008405f19dd83247405392/subsys/testsuite/ztest/include/zephyr/ztest_test_new.h#L370-L379

Thanks to that, this testcase will be marked as SKIP when there is no `CONFIG_SECURE_BOOT` enabled. Without marking it explicate as skip, this test will be marked as PASS each time when there is no `SECURE_BOOT`, what could be misleading.